Devs/felipealmeida/evas fixes

JPTIZ opened the pull request.

joaoantoniocardoso committed:
Enabling ecore_win32+ecore_input

joaoantoniocardoso committed:
Addming a missing ole2 header

joaoantoniocardoso committed:
Fixing pid_t undefined for windows

joaoantoniocardoso committed:
Fixing headers after unposix removal

JPTIZ committed:
build: Fallback libjpeg to wrap

felipealmeida committed:
win32: Fix compilation with openssl and jpeg

felipealmeida committed:
win32: Fix unistd.h include and incdir

felipealmeida committed:
win32: eet image inclusion order because of jpeg typedefs

felipealmeida committed:
win32: emile compilation with win32 and jpeg inclusion order

felipealmeida committed:
win32: Add asprintf version

felipealmeida committed:
evil: Add dirname to evil

felipealmeida committed:
win32: Fix includes for unistd.h and ligen.h for Windows

felipealmeida committed:
win32: Add lua to subprojects

felipealmeida committed:
win32: Fix include and remove from compilation to win32

felipealmeida committed:
win32: Move Eina_Error error code definitions to appropriate library

felipealmeida committed:
win32: Add EVAS_BUILD evas_opt

felipealmeida committed:
win32: Fix LIB_API definitions

felipealmeida committed:
win32: Allow use of wrap dependency in jpeg

felipealmeida committed:
win32: Change winsock.h for winsock2.h as not to conflict

felipealmeida committed:
win32: Add eet dependency  to vg_common

felipealmeida committed:
win32: Remove unistd.h and libgen.h from tests

felipealmeida committed:
evil: Add gettimeofday implementation

felipealmeida committed:
win32: Disable test that uses O_CREAT and other flags temporarily

felipealmeida committed:
win32: Fix lua wrap

felipealmeida committed:
win32: Add jpeg wrap

felipealmeida committed:
win32: Fix test using direct pipe functions

felipealmeida committed:
win32: Avoid skipping running callbacks because timeout is 0

felipealmeida committed:
win32: gettimeofday is always available now

felipealmeida committed:
win32: special case for MSVC compilation without lib prefix in modules

felipealmeida committed:
win32: Remove unnecessary delay for evil startup

felipealmeida committed:
win32: Fix pipe calls in ecore test

felipealmeida committed:
win32: Max count must be greater than 0

felipealmeida committed:
win32: Use appropriate prefixed pipe_ functions

joaoantoniocardoso committed:
Using compile_args to be able to use absolute path for libcrypto

felipealmeida committed:
win32: Remove multiple frees for stringshare

felipealmeida committed:
win32: Remove platform specific headers and make compatible with Windows compilation

felipealmeida committed:
win32: Remove platform specific headers and make compatible with Windows compilation

felipealmeida committed:
win32: Make evas examples compilable by MSVC and GUI applications on Windows

felipealmeida committed:
win32: Remove need to use script to rename to module.ext for modules

felipealmeida committed:
win32: Enable compilation of examples

felipealmeida committed:
win32: Fix signed overflow with bitsets and enumerations

felipealmeida committed:
win32: Remove unistd.h headers from examples

They are already included by Eina.h in the right platforms

felipealmeida committed:
win32: Add special case for windows font path

felipealmeida committed:
win32: Add special case for Windows absolute path in font pah

felipealmeida committed:
Enable new dependencies

felipealmeida committed:
win32: Add -in-tree option in edje_cc for Windows

felipealmeida committed:
win32: Remove and replace platform specific headers

felipealmeida committed:
win32: Fix pending_threads calculation inside loop and avoid infinite loop

felipealmeida committed:
unistd.h

felipealmeida committed:
win32: Do not fail if we can't find libcurl

felipealmeida committed:
clang: Fix enumeration signed overflow

felipealmeida committed:
win32: Fix eina modules search with meson install

felipealmeida committed:
elm: Add evil and pcre dependency to prefs

felipealmeida committed:
win32: Rename elementary_test lib to not clash with executable in Windows

felipealmeida committed:
win32: Make emotion optional

joaoantoniocardoso commented:
> Wouldn't it be better to surround it with `#ifndef _MSC_VER`?

They are not really used.

felipealmeida committed:
win32: Make ethumb optional

felipealmeida committed:
win32: Make eldbus and atspi be optional

felipealmeida committed:
edje: Only compile svg edje if svg loader is enabled

felipealmeida committed:
win32: Remove file system options that do not exist in Windows

felipealmeida committed:
eina: Do not crash when parameter NULL is passed to eina_thread_cancel

felipealmeida committed:
win32: Do not use quicklaunch with MSVC

felipealmeida committed:
win32: Enable libpng dependency in windows

felipealmeida committed:
win32: Avoid compiling examples that use platform-specific functions not available in Windows

felipealmeida committed:
win32: Enable png and eet loaders in configure.bat

felipealmeida committed:
elm: Use platform neutral embed_config C program instead of script

felipealmeida committed:
elm: Remove declaration from wrong library

_efl_startup_time is declared in Ecore library. It shouldn't be
declared in Elementary because it causes mismatches with importing and
exporting of symbol by MSVC's linker.

felipealmeida committed:
evil: Add localtime_r implementation using localtime

felipealmeida committed:
win32: Do not compile deprecated efreet libraries in Windows

felipealmeida committed:
win32: Do not compile kde legacy initialization in Windows

felipealmeida committed:
win32: Module macros are always export

felipealmeida committed:
win32: Define EMODAPI_WEAK for Windows

felipealmeida committed:
win32: Rename regex.h to pcreposix.h

felipealmeida committed:
win32: Temporary libpng wrap dependency

felipealmeida committed:
win32: Fix eldbus dependency (squash with other dbus later)

felipealmeida committed:
win32: Embed_config handle CRLF too

felipealmeida committed:
evil: Fix fcntl for F_SETLK and F_SETLKW wrong length calculation

felipealmeida committed:
win32: Add EINA_LOG_LEVEL to build command

joaoantoniocardoso commented:
To maintain track of what is done here, at this time all the commits except the followings are already merged into native-windows:
  - 2f3c8295db3e1fa9c62462dbeb6057085e19f87e -> #380,
  - 426386578bb6f4e045474dbf1d94a502f29ff18b,
  - 63c60ade535bcc2e7dcac8e87a2b31c3cd2a3ee7,
  - 9368908de1a77b73afe99af8a5ee9e4773446dfb,
  - 0abb93b2dedb3784cc302925e0967cf78d00c157,
